test/libyul/yulOptimizerTests/fullSuite/medium.yul

```diff
@@ -21,10 +21,17 @@
 // ----
 // {
 //     {
-//         let _1 := mload(0x40)
-//         mstore(0x40, add(_1, 0x20))
-//         mstore(0x40, add(_1, 96))
-//         mstore(add(_1, 128), 2)
+//         let p := mload(0x40)
+//         mstore(0x40, add(p, 0x20))
+//         mstore(0x40, add(p, 96))
+//         mstore(add(p, 128), 2)
+//         let _1 := 1
 //         mstore(0x40, 0x20)
+//         for { } _1 { }
```

Comment:
Oh come on!

Reply:
That's probably even not so easy to get rid of...

Reply:
We need #7406

Reply:
Actually this is even wrong, isn't it? Originally this was ``for { switch mul(1,2) case 2 { mstore(0x40, 0x20) } } sub(1,1) {} { mstore(0x80, 0x40) }``, right? Which has a condition of ``sub(1,1)``, so it should never be executed - yet after optimization it *is* executed once...

Reply:
I think the old and new version of this are not equivalent - the old version didn't have the write to 128, which goes along with the original code, but the new version does have the write...

Reply:
I also spent 10 minutes thinking about this earlier today, but it is correct. It first enters the loop but then breaks again immediately.

Reply:
It is not optimized away because `_1` is re-assigned below, even though it is dead code.

Reply:
Ah right, yeah, it's true, it is correct :-).